js/src/components/google-combo-account-card/claim-ads-account/claim-ads-account.js

```diff
@@ -0,0 +1,86 @@
+/**
+ * External dependencies
+ */
+import { useCallback, useState } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import ClaimAccountButton from '.~/components/google-ads-account-card/claim-account-button';
+import Section from '.~/wcdl/section';
+import useGoogleAdsAccountStatus from '.~/hooks/useGoogleAdsAccountStatus';
+import { useAppDispatch } from '.~/data';
+import useGoogleAdsAccount from '.~/hooks/useGoogleAdsAccount';
+import useWindowFocusCallbackIntervalEffect from '.~/hooks/useWindowFocusCallbackIntervalEffect';
+import './claim-ads-account.scss';
+
+/**
+ * ClaimAdsAccount component.
+ *
+ * @return {JSX.Element} ClaimAdsAccount component.
+ */
+const ClaimAdsAccount = () => {
+	const [ updating, setUpdating ] = useState( false );
+	const { googleAdsAccount } = useGoogleAdsAccount();
+	const { hasAccess } = useGoogleAdsAccountStatus();
+	const { fetchGoogleAdsAccountStatus } = useAppDispatch();
+
+	const shouldClaimGoogleAdsAccount = Boolean(
+		googleAdsAccount?.id && hasAccess === false
+	);
+
+	const checkUpdatedAdsAccountStatus = useCallback( async () => {
+		if ( ! shouldClaimGoogleAdsAccount ) {
+			return;
+		}
+
+		await fetchGoogleAdsAccountStatus();
+	}, [ fetchGoogleAdsAccountStatus, shouldClaimGoogleAdsAccount ] );
+
+	useWindowFocusCallbackIntervalEffect( checkUpdatedAdsAccountStatus, 30 );
+
+	const handleOnClick = () => {
```

Comment:
One observation when testing this is that if someone clicks the Claim... button but then does not accept the invite, it might be nice to reset this value on the next `checkUpdatedAdsAccountStatus()` check so they could click the button to open the modal again. Otherwise, the have to refresh the page to get the button to work again.

Reply:
@joemcgill I think that'll be confusing for the user 🤔 Once the user clicked the claim button, he will see the "Updating..." text for the button. Then shortly he will see the claim button back. The user might think there's an issue and will want to click the button again. Am not sure, I just feel it'll be weird for the user.